kong/tools/mime_type.lua

```diff
@@ -106,40 +106,21 @@ local function includes(this, other)
     error("other must be a table", 2)
   end
 
-  local this_type = this.type
-  local this_subtype = this.subtype
-  local other_type = other.type
-  local other_subtype = other.subtype
-
-  if this_type == WILDCARD then
+  if this.type == WILDCARD then
     -- */* includes anything
     return true
   end
 
-  if this_type == other_type then
-    if this_subtype == other_subtype or this_subtype == WILDCARD then
+  if this.type == other.type then
+    if this.subtype == other.subtype or this.subtype == WILDCARD then
       return true
     end
 
-    if sub(this_subtype, 1, 2) == "*+" then
-      -- subtype is wildcard with suffix, e.g. *+json
-      local this_subtype_suffix = sub(this.subtype, 3)
-      local pattern = "^.-+" .. this_subtype_suffix ..  "$"
-      if find(other_subtype, pattern) then
+    -- considering included when this_subtype does not contain a suffix and is the suffix of other_subtype
+    if not find(this.subtype, "+", nil, true) then -- this_subtype does not contain suffix
+      if match(other.subtype, this.subtype .. "$") then -- suffix match
```

Comment:
```suggestion
      if match(other.subtype, "+" .. this.subtype .. "$") then -- suffix match
```
We can be sure that we're not looking at an exact match at this point, but we should guard against matching unrelated suffixes.  Do we need to quote the `+`?